src/components/postgresConnect.ts

```diff
@@ -316,6 +316,82 @@ class PostgresDB {
       return false
     }
   }
+
+  public getTemplateStages = async (templateId: number) => {
+    const text = 'SELECT id, number, title FROM template_stage WHERE template_id = $1'
+    try {
+      const result = await this.query({ text, values: [templateId] })
+      return result.rows
+    } catch (err) {
+      console.log(err.message)
+      throw err
+    }
+  }
+
+  public getCurrentStageHistory = async (applicationId: number) => {
+    const text =
+      'SELECT id, stage_id FROM application_stage_history WHERE application_id = $1 and is_current = true'
```

Comment:
Good point.

Reply:
I think the reason I didn't was because the current `application_stage_status` view doesn't include the id's and stuff, which we need for creating the subsequent status_history record.

I've made another view now called `application_stage_status` that has all the relevant fields from the `stage`, `application_stage_history` and `application_status_history` tables, which will be more useful for querying.

Reply:
> And this one could also return the current status altogether.

Only if they're correctly linked (i.e every current status history is linked to a current stage history), but if they're not, getting them both together gets very complicated.